<?xml version="1.0"?>

<launch>
    <arg name="scout_sim_flg" default="true"/>
    <group if="$(eval arg('scout_sim_flg') == false)">
        <node pkg="wpt_manager" type="wpt_manager_node" name="wpt_manager_node" output="screen" respawn="false">
            <param name="conops_rviz" type="string" value="conops_rviz1" />
            <param name="random_generation" type="bool" value="false" />
            <param name="predefined_wpt_file_path" type="string" value="$(find wpt_manager)/wpt_file/test.csv" />
            <param name="pub_wpt_topic_name" type="string" value="/scout/behaviorNgoal" />
            <param name="sub_odom_topic_name" type="string" value="/odom_ndt_global_prediction" />
            <param name="sub_rviz_goal_topic_name" type="string" value="/move_base_simple/goal" />
            <param name="sub_rviz_conops_topic_name" type="string" value="/scout/behaviorNgoalArray" />
            <param name="max_num_of_wpt" type="int" value="100" />
            <param name="num_of_random_wpt" type="int" value="10" />
            <param name="goal_arrived_dist" type="double" value="1.5" />
            <!-- <param name="random_generation_coordination" type="string" value="inBodyCoordinate" /> -->
            <param name="random_generation_coordination" type="string" value="inGlobalCoordinate" />
            <!--param name="random_generation_coordination" type="string" value="conops" /-->

            <param name="random_generation_body_x_min" type="double" value="4.0" />
            <param name="random_generation_body_x_max" type="double" value="10.0" />
            <param name="random_generation_body_y_min" type="double" value="-10.0" />
            <param name="random_generation_body_y_max" type="double" value="10.0" />
            <param name="random_generation_body_z_min" type="double" value="-0.5" />
            <param name="random_generation_body_z_max" type="double" value="0.5" />
            <param name="predefined_altitude_limit" type="double" value="2.5" />

            <param name="random_generation_global_x_min" type="double" value="0.0" />
            <param name="random_generation_global_x_max" type="double" value="100.0" />
            <param name="random_generation_global_y_min" type="double" value="-30.0" />
            <param name="random_generation_global_y_max" type="double" value="30.0" />
            <param name="random_generation_global_z_min" type="double" value="1.0" />
            <param name="random_generation_global_z_max" type="double" value="2.5" />
            
            <param name="conop_takeoff_z" type="double" value="1.5" />
            <param name="conop_flight_fixed_z" type="double" value="1.5" />
        </node>
    </group>
    <group if="$(eval arg('scout_sim_flg') == true)">
        <node pkg="wpt_manager" type="wpt_manager_node" name="wpt_manager_node" output="screen" respawn="false">
            <param name="conops_rviz" type="string" value="conops_rviz_not" />
            <param name="random_generation" type="bool" value="false" />
            <param name="predefined_wpt_file_path" type="string" value="$(find wpt_manager)/wpt_file/test.csv" />
            <param name="pub_wpt_topic_name" type="string" value="/scout/behaviorNgoal" />
            <param name="sub_odom_topic_name" type="string" value="/scout/ground_truth/state_raw" />
            <param name="sub_rviz_goal_topic_name" type="string" value="/move_base_simple/goal" />
            <param name="sub_rviz_conops_topic_name" type="string" value="/scout/behaviorNgoalArray" />
            <param name="max_num_of_wpt" type="int" value="100" />
            <param name="num_of_random_wpt" type="int" value="10" />
            <param name="goal_arrived_dist" type="double" value="1.5" />
            <!-- <param name="random_generation_coordination" type="string" value="inBodyCoordinate" /> -->
            <param name="random_generation_coordination" type="string" value="inGlobalCoordinate" />
            <!--param name="random_generation_coordination" type="string" value="conops" /-->

            <param name="random_generation_body_x_min" type="double" value="4.0" />
            <param name="random_generation_body_x_max" type="double" value="10.0" />
            <param name="random_generation_body_y_min" type="double" value="-10.0" />
            <param name="random_generation_body_y_max" type="double" value="10.0" />
            <param name="random_generation_body_z_min" type="double" value="-0.5" />
            <param name="random_generation_body_z_max" type="double" value="0.5" />
            <param name="predefined_altitude_limit" type="double" value="2.5" />

            <param name="random_generation_global_x_min" type="double" value="0.0" />
            <param name="random_generation_global_x_max" type="double" value="100.0" />
            <param name="random_generation_global_y_min" type="double" value="-30.0" />
            <param name="random_generation_global_y_max" type="double" value="30.0" />
            <param name="random_generation_global_z_min" type="double" value="1.0" />
            <param name="random_generation_global_z_max" type="double" value="2.5" />
            
            <param name="conop_takeoff_z" type="double" value="1.5" />
            <param name="conop_flight_fixed_z" type="double" value="1.5" />
        </node>
        <!-- <node pkg="tf" type="static_transform_publisher" name="kaist_tf_broadcaster" args="0 0 0 0 0 0 /scout1/base_link /scout1/velodyne 10" /> -->

    </group>
</launch>
